functions/required.go

```diff
@@ -0,0 +1,14 @@
+package functions
+
+import "errors"
+
+func required(msg string, val interface{}) (interface{}, error) {
+	if val == nil {
+		return val, errors.New(msg)
+	} else if _, ok := val.(string); ok {
+		if val == "" {
+			return val, errors.New(msg)
+		}
+	}
+	return val, nil
+}
```

Comment:
https://codebeat.co/projects/github-com-julienbreux-clamp-master/pull_requests/394947

```suggestion
func required(msg string, val interface{}) (interface{}, error) {
	if val == nil {
		return val, errors.New(msg)
	}
	if _, ok := val.(string); ok && val == "" {
		return val, errors.New(msg)
	}
	return val, nil
}
```

Reply:
Oh... My bad, I should have spotted that early on.